ambry-server/src/integration-test/java/com.github.ambry.server/VcrRecoveryTest.java

```diff
@@ -47,106 +65,224 @@
+   * @throws Exception If an exception happens.
+   */
+  @Test
+  public void basicCloudRecoveryTest() throws Exception {
+    // Create blobs and upload to cloud destination.
+    int userMetaDataSize = 100;
+    byte[] userMetadata = new byte[userMetaDataSize];
+    TestUtils.RANDOM.nextBytes(userMetadata);
+    Map<BlobId, Integer> blobIdToSizeMap = new HashMap<>();
     for (BlobId blobId : blobIds) {
+      int blobSize = Utils.getRandomShort(TestUtils.RANDOM);
+      byte[] data = new byte[blobSize];
+      BlobProperties blobProperties =
+          new BlobProperties(blobSize, "serviceid1", null, null, false, Utils.Infinite_Time, accountId, containerId,
+              false, null);
+      TestUtils.RANDOM.nextBytes(data);
+      blobIdToSizeMap.put(blobId, blobSize);
+      PutMessageFormatInputStream putMessageFormatInputStream =
+          new PutMessageFormatInputStream(blobId, null, blobProperties, ByteBuffer.wrap(userMetadata),
+              new ByteBufferInputStream(ByteBuffer.wrap(data)), blobSize);
       long time = System.currentTimeMillis();
       CloudBlobMetadata cloudBlobMetadata =
-          new CloudBlobMetadata(blobId, time, Utils.Infinite_Time, blobSize, CloudBlobMetadata.EncryptionOrigin.NONE);
-      latchBasedInMemoryCloudDestination.uploadBlob(blobId, blobSize, cloudBlobMetadata,
-          new ByteArrayInputStream(data));
+          new CloudBlobMetadata(blobId, time, Utils.Infinite_Time, putMessageFormatInputStream.getSize(),
+              CloudBlobMetadata.EncryptionOrigin.NONE);
+      latchBasedInMemoryCloudDestination.uploadBlob(blobId, putMessageFormatInputStream.getSize(), cloudBlobMetadata,
+          putMessageFormatInputStream);
     }
 
-    // start ambry server with data node
-    recoveryCluster.initializeServers(notificationSystem, vcrNode, recoveryProperties);
-    recoveryCluster.startServers();
+    // Waiting for download attempt
+    assertTrue("Did not recover all blobs in 2 minutes",
+        latchBasedInMemoryCloudDestination.awaitDownload(2, TimeUnit.MINUTES));
+
+    // Waiting for replication to complete
+    Thread.sleep(10000);
 
-    //Waiting for recovery
+    // Test recovery by sending get request to recovery node
+    testGetOnRecoveryNode(blobIdToSizeMap);
+  }
+
+  /**
+   * Test recovery from one vcr node to one disk based data node for large blobs.
+   * Creates a vcr node and a disk based data node. Uploads data to vcr node and verifies that they have been downloaded.
+   * @throws Exception If an exception happens.
+   */
+  @Test
+  public void cloudRecoveryTestForLargeBlob() throws Exception {
+    // Create blobs and upload to cloud destination.
+    int userMetaDataSize = 100;
+    byte[] userMetadata = new byte[userMetaDataSize];
+    TestUtils.RANDOM.nextBytes(userMetadata);
+    Map<BlobId, Integer> blobIdToSizeMap = new HashMap<>();
+    int blobSize = FOUR_MB_SZ;
```

Comment:
So, we use 4MB to mock large blob, right? I guess this is biggest size for a chunk from ambry server's point of view. Maybe adding some comment about why we choose 4MB would be better.